sites/geohub/src/components/App.svelte

```diff
@@ -1,22 +1,61 @@
 <script lang="ts">
+  import { fade } from 'svelte/transition'
   import type { Map as MaplibreMap } from 'maplibre-gl'
   import { Split } from '@geoffcox/svelte-splitter/src'
   import Header from '$components/Header.svelte'
-  import BannerMessageControl from '$components/BannerMessageControl.svelte'
   import Map from '$components/Map.svelte'
+  import { Button } from '@undp-data/svelte-undp-design'
   import Content from './Content.svelte'
-  import { map as mapStore, indicatorProgress } from '$stores'
+  import { map as mapStore, layerList, isStyleEdited } from '$stores'
   import SplitterControl from '$components/SplitterControl.svelte'
+  import { beforeNavigate, goto } from '$app/navigation'
+  import Notification from '$components/controls/Notification.svelte'
+  import { page } from '$app/stores'
 
   let map: MaplibreMap
   let headerHeight: number
   let isMenuShown = true
-
   let innerWidth: number
   let innerHeight: number
+  let dialogOpen = false
+  let cancel: () => void
+  let toURL: URL
+  let isContinueButtonClicked = false
   let settings = {
     sideBarPosition: 'left',
   }
+
+  const openConfirmationModal = () => {
+    dialogOpen = true
+  }
+
+  const handleCancel = () => {
+    dialogOpen = false
+  }
+  const handleKeyDown = (e) => {
+    if (e.key === 'Escape') {
+      handleCancel()
+    }
+  }
+
+  const handleContinue = () => {
+    isContinueButtonClicked = true
+    layerList.update(() => [])
+    goto(toURL.pathname)
+    dialogOpen = false
+  }
+
+  beforeNavigate(({ cancel, to }) => {
+    if (isContinueButtonClicked || $layerList.length === 0 || !isStyleEdited) {
```

Comment:
For saved map, please check the following conditions to detect whether the style.json is modified or not.

- $map.getStyle().sources !== page.data.style.sources (you should check whether source URL is the same or not)
- $map.getStyle().layers.length !== $page.data.style.layers.length
- $layerList !== $page.data.layers (you have to check all properties for each layer style object)